### PR TITLE
#150749930 Paystack Integration

### PR DESCRIPTION
**Pull request by noordean**
#### What does this PR do?
It integrates paystack payment system
#### Description of Task to be completed?
Integrate paystack payment system
#### How should this be manually tested?
By running `npm run test-local`
#### Any background context you want to provide?
Ensure selenium server and chromedriver are installed and make sure admin details are set in the .env file.
#### What are the relevant pivotal tracker stories?
#150749930
#### Screenshots (if appropriate)
screen shot 2017-09-21 at 8 15 40 am

screen shot 2017-09-21 at 8 16 12 am

screen shot 2017-09-21 at 8 18 06 am

screen shot 2017-09-21 at 8 16 31 am

#### Questions:
Nil

### CODE REVIEW

File: tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+const expect = require("chai").expect;
+const getId = require("../../../lib/get-elements.js");
+const dotenv = require("dotenv");
+
+dotenv.config();
+beforeEach(function () {
+  const browserConfig = yaml.safeLoad(fs.readFileSync("./tests/acceptance-tests/config/settings.yml", "utf8"));
+  const baseUrl = browserConfig.base_url.toString();
+  browser.url(baseUrl);
+  // browser.getSession().then(function (sessionid) {
```
Comment (temilaj) [inline tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: why do you have commented-out code in source control?

File: tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    browser.click("//span[text()='Paystack Payment']");
+    browser.waitForExist("input[name='payerEmail']");
+    browser.setValue("input[name='payerEmail']", guestEmail);
+    browser.setValue(".cc-number", process.env.cardNumber);
+    browser.setValue("#expiry", process.env.expTime);
+    browser.setValue("#cvv", process.env.cv);
+    browser.click("#pay-btn");
+    browser.waitForExist(".order-item");
+    expect(browser.getAttribute("div", "order-item")).to.exist;
+  });
```
Comment (temilaj) [inline tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: Is this test complete?. How do we know that the payment will be successful

File: imports/plugins/included/paystack/lib/api/paystack.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import request from "request";
+
+export const Paystack = {};
+
+const paystackHeaders = (secret) => {
```
Comment (temilaj) [inline imports/plugins/included/paystack/lib/api/paystack.js]: Consider improving the documentation in your PR. Almost all funcitons do not have any form of documentation

File: tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js (hunk 1 above)
Comment (noordean) [reply tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: I will fix that right now.

File: tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js (hunk 2 above)
Comment (noordean) [reply tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: If the payment is successful, a div element with class "order-item" gets rendered, it shows a list of ordered products. So without the payment being successful, there is no way the element would exist

File: imports/plugins/included/paystack/lib/api/paystack.js (hunk 3 above)
Comment (noordean) [reply imports/plugins/included/paystack/lib/api/paystack.js]: I will fix this right now

File: tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js (hunk 2 above)
Comment (temilaj) [reply tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: Do you think it'll be valuable to check what's inside the div?

Comment (noordean) [reply tests/acceptance-tests/test/specs/smoke-tests/paystack.app-test.js]: It's not checking for what is inside the div. It's checking if a div that has 'order-item' in one of its attributes exists. So it's checking for the existence of the div

**Pull request merged**